pkg/model/api/v1/dashboard.go

```diff
@@ -193,16 +195,16 @@ func (d *Dashboard) validate() error {
 	if reflect.DeepEqual(d.Spec, DashboardSpec{}) {
 		return fmt.Errorf("spec cannot be empty")
 	}
-	return d.verifyAndSetJSONReferences()
+	return verifyAndSetJSONReferences(d.Spec.Layouts, d.Spec.Panels)
 }
 
 // verifyAndSetJSONRef will check that each JSON Reference are pointing to an existing object and will set the related pointer in the JSONRef.Object
-func (d *Dashboard) verifyAndSetJSONReferences() error {
-	for _, layout := range d.Spec.Layouts {
+func verifyAndSetJSONReferences(layouts []dashboard.Layout, panels map[string]*Panel) error {
```

Comment:
NB function signatures modified to allow reuse by ephemeral dashboards

Reply:
can´t you change the signature to something belonging to DashboardSpec ? As you are already reusing DashboardSpec into the ephemeralDashboard. It would imply less changes. 

```go
func (d *DashboardSpec) verifyAndSetJSONReferences() error
```